flexmeasures/utils/tests/test_time_utils.py

```diff
@@ -41,3 +42,37 @@ def test_naturalized_datetime_str(
         h_ago = None
     print(h_ago)
     assert naturalized_datetime_str(h_ago, now=now) == exp_result
+
+
+@pytest.mark.parametrize(
+    "window_size,now,exp_start,exp_end",
+    [
+        (
+            5,
+            datetime(2021, 4, 30, 15, 1),
+            datetime(2021, 4, 30, 14, 55),
+            datetime(2021, 4, 30, 15),
+        ),
+        (
+            15,
+            datetime(2021, 4, 30, 3, 36),
+            datetime(2021, 4, 30, 3, 15),
+            datetime(2021, 4, 30, 3, 30),
+        ),
+        (
+            10,
+            datetime(2021, 4, 30, 0, 5),
+            datetime(2021, 4, 29, 23, 50),
+            datetime(2021, 4, 30, 0, 0),
+        ),
+    ],
+)
```

Comment:
Please add a test for the boundary condition. Like one of (5, 10:05, 10:00, 10:05) or (5, 10:05, 9:55, 10:00), depending on what you expect (I guess the former).

Reply:
More boundary conditions:

(60, January 1st 0:00, ..., ...) # new year
(60, March 28th 2021 3:00, ..., ...) # DST transition